test/qa/test_routing.py

```diff
@@ -27,86 +29,145 @@ def setup_function(function):
+            print(sh.nordvpn.connect())
+            assert network.is_available()
+
             lib.set_routing("off")
-            routes = sh.ip.route.show.table(table)
-            rules = sh.ip.rule()
-            assert not "nordlynx" in routes
-            assert not "mark" in rules
-            assert network.is_not_available()
-        
-        lib.set_routing("on")
-        routes = sh.ip.route.show.table(table)
-        rules = sh.ip.rule()
-        assert "nordlynx" in routes
-        assert "mark" in rules
-        assert network.is_available()
-
-    print(sh.nordvpn.disconnect())
+            assert not settings.get_is_routing_enabled()
+            assert not network_interface in sh.ip.route.show.table(TABLE)
+            assert not "mark" in sh.ip.rule()
+            assert not network.is_available()
+
+            lib.set_routing("on")
+            assert settings.get_is_routing_enabled()
+            assert network_interface in sh.ip.route.show.table(TABLE)
+            assert "mark" in sh.ip.rule()
+            assert network.is_available()
+
+
+@pytest.mark.skip("LVPN-3273; LVPN-1574")
+@pytest.mark.parametrize("tech,proto,obfuscated", lib.TECHNOLOGIES)
+@pytest.mark.flaky(reruns=2, reruns_delay=90)
+@timeout_decorator.timeout(40)
+def test_connected_routing_enable_disable(tech, proto, obfuscated):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
+
+    network_interface = "nordtun" if tech == "openvpn" else "nordlynx"
+
+    lib.set_routing("off")
+    assert not settings.get_is_routing_enabled()
+
+    with lib.Defer(sh.nordvpn.set.routing.on):
+        with lib.Defer(sh.nordvpn.disconnect):
+            print(sh.nordvpn.connect())
+            assert not network.is_available()
+
+            lib.set_routing("on")
+            assert settings.get_is_routing_enabled()
+            assert network_interface in sh.ip.route.show.table(TABLE)
+            assert "mark" in sh.ip.rule()
+            assert network.is_available()
+
+            lib.set_routing("off")
+            assert not settings.get_is_routing_enabled()
+            assert not network_interface in sh.ip.route.show.table(TABLE)
+            assert not "mark" in sh.ip.rule()
+            assert not network.is_available()
+
+
+@pytest.mark.parametrize("tech,proto,obfuscated", lib.TECHNOLOGIES_BASIC1)
+def test_meshnet_on_routing_disable(tech, proto, obfuscated):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
+
+    with lib.Defer(sh.nordvpn.set.mesh.off):
+        sh.nordvpn.set.mesh.on()
+        assert "Routing is currently used by Meshnet. Disable it first." in sh.nordvpn.set.routing.off()
+        assert settings.get_is_routing_enabled()
+
+
+@pytest.mark.parametrize("tech,proto,obfuscated", lib.TECHNOLOGIES)
+def test_routing_already_enabled(tech, proto, obfuscated):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
+
+    assert "Routing is already set to 'enabled'." in sh.nordvpn.set.routing.on()
```

Comment:
who is setting the routing here?

Reply:
We are setting routing to enabled with `sh.nordvpn.set.routing.on()` in `assert "Routing is already set to 'enabled'." in sh.nordvpn.set.routing.on()`